src/query.ts

```diff
@@ -36,12 +36,15 @@ export class Query {
         // 1. Replace tag:#abc by "tag:#abc" because this parser doesn't like the '#' symbol
         // 2. Replace path:"abc def/ghi" by "path:abc def/dhi" because the parser doesn't like quotes as part of the words
         // 3. Same goes for linkedto:"", linkedfrom:"" and name:""
+        // 4. Replace ["property":"value"] with single quotes to avoid parser complaints.
         let newString = queryString
             .replace(regex.TAG_NAME_WITH_HEADER_AND_WILDCARD, '"tag:$1"')
             .replace(regex.PATH_QUERY_WITH_HEADER, '"path:$1"')
             .replace(regex.LINKEDTO_QUERY_WITH_HEADER, '"linkedto:$1"')
             .replace(regex.LINKEDFROM_QUERY_WITH_HEADER, '"linkedfrom:$1"')
-            .replace(regex.NAME_QUERY_WITH_HEADER, '"name:$1"');
+            .replace(regex.NAME_QUERY_WITH_HEADER, '"name:$1"')
+            .replace(/^\[(")(.+?)\1:/, "['$2':")
+            .replace(/:(")(.+)?\1\]/, ":'$2']");
```

Comment:
This is admittedly terrible; we may be stretching the limits of `boon-js` at this point, but I don't see a better way to handle it.